### PR TITLE
Fixups for CMP tests

### PR DESCRIPTION
**Pull request by DDvO**
This PR has evolved into a collection of fixups for various minor loose ends in CMP tests.

* 81-test_cmp_cli.t: Stop unlinking test output files according to #11080
* 81-test_cmp_cli: Make test output files all different according to #11080
* test/recipes/81-test_cmp_cli_data/Mock/server.cnf: minor cleanup
* test/cmp_{client,msg}_test.c: minor code cleanup

### CODE REVIEW

File: test/recipes/81-test_cmp_cli.t [hunk 1]
```diff
@@ -200,7 +200,7 @@ sub test_cmp_cli_aspect {
             }
         }
     };
-    unlink "test.cert.pem", "test.cacerts.pem", "test.extracerts.pem";
+    # not unlinking "test.cert.pem", "test.cacerts.pem", "test.extracerts.pem"
```
Comment (levitte) [inline test/recipes/81-test_cmp_cli.t:203]: I would suggest ensuring that each case outputs different files as well.  It's fine if you want to do that in a separate PR

Comment (DDvO) [reply test/recipes/81-test_cmp_cli.t]: > I would suggest ensuring that each case outputs different files as well. It's fine if you want to do that in a separate PR

Done in a further commit of this PR.

**Comment by DDvO** [comment]
Updating the CMP CLI tests I also found and fixed mem leaks on error in `load_certs()` and `load_crls()` of `apps.c`.

File: test/recipes/81-test_cmp_cli.t (hunk 1 above)
Comment (levitte) [reply test/recipes/81-test_cmp_cli.t]: Uhmmm, did you push that commit?

Comment (DDvO) [reply test/recipes/81-test_cmp_cli.t]: Yes, the commit differentiating the output files is commit bd185484de5fa1564d32786eb480654e5fe990b3.

Comment (DDvO) [reply test/recipes/81-test_cmp_cli.t]: Its main changes are in `test/recipes/81-test_cmp_cli_data/test_enrollment.csv`
while the changes to other files are minor test case improvements I came across on this occasion.

File: doc/man3/OSSL_CMP_CTX_new.pod [hunk 2]
```diff
@@ -97,9 +97,9 @@ OSSL_CMP_CTX_set1_senderNonce
                                       const X509_NAME *name);
  int OSSL_CMP_CTX_set0_trustedStore(OSSL_CMP_CTX *ctx, X509_STORE *store);
  X509_STORE *OSSL_CMP_CTX_get0_trustedStore(const OSSL_CMP_CTX *ctx);
- int OSSL_CMP_CTX_set1_untrusted_certs(OSSL_CMP_CTX *ctx,
+ int OSSL_CMP_CTX_set1_untrusted(OSSL_CMP_CTX *ctx,
                                        STACK_OF(X509) *certs);
```
Comment (slontis) [inline doc/man3/OSSL_CMP_CTX_new.pod:101]: indentation is off.

File: doc/man1/openssl-cmp.pod.in [hunk 3]
```diff
@@ -601,27 +608,32 @@ is typically used when authenticating with pre-shared key (password-based MAC).
 
 =item B<-secret> I<arg>
 
-Source of secret value to use for creating PBM-based protection of outgoing
-messages and for verifying any PBM-based protection of incoming messages.
+Prefer PBM-based message protection with given source of secret value.
+This useed for creating PBM-based protection of outgoing messages and as far
```
Comment (slontis) [inline doc/man1/openssl-cmp.pod.in:612]: This useed for ??

File: apps/lib/apps.c [hunk 4]
```diff
@@ -806,6 +822,9 @@ int load_key_certs_crls(const char *uri, int maybe_stdin,
 
  end:
     OSSL_STORE_close(ctx);
+    if (failed != NULL)
+        return 0;
```
Comment (slontis) [inline apps/lib/apps.c:826]: This no longer calls ERR_print_errors() if it returns here

File: doc/man3/OSSL_CMP_CTX_new.pod (hunk 2 above)
Comment (DDvO) [reply doc/man3/OSSL_CMP_CTX_new.pod]: Oops, fixed.

File: doc/man1/openssl-cmp.pod.in (hunk 3 above)
Comment (DDvO) [reply doc/man1/openssl-cmp.pod.in]: Fixed this typo and further improved the sentence.

File: apps/lib/apps.c (hunk 4 above)
Comment (DDvO) [reply apps/lib/apps.c]: > This no longer calls ERR_print_errors() if it returns here

Good point, and this was missing also in case of `return 0` on out-of-memory. Fixed.

**Comment by DDvO** [comment]
Thanks @slontis for your comments. 
Anything more to do here?

**Comment by DDvO** [comment]
I've just made separate PRs out of those parts of this PR that are (likely) to be considered an API change
such that the remaining commits in this PR are not urgent to be handled before feature freeze.

**Comment by DDvO** [comment]
I've just been carving out most of this PR into separate ones - thanks @t-j-h for already approving most of these!

**Comment by DDvO** [comment]
What remains here is just a couple of fixup for the CMP tests.

**Review by t-j-h — APPROVED** [review]

**Comment by t-j-h** [comment]
Sorted out. Splitting these into logical chunks makes the review process a lot smoother.
As always, these improvements are very much appreciated!

**Comment by DDvO** [comment]
> Sorted out. Splitting these into logical chunks makes the review process a lot smoother.
> As always, these improvements are very much appreciated!

Very pleased @t-j-h about your positive feedback!

**Comment by DDvO** [comment]
Merged as well - thanks also @levitte and @slontis for your comments.
